### DISCUSSION TITLE
Next.js 13 and Server Components

### DISCUSSION THREAD

**Discussion by cobrani** [wpengine/faustjs#1128] [Ideas]
Hi all,
first of all thank you for the great work on the new faustjs using Apollo!
Maybe it's still a little bit early, but Vercel introduced Nextjs 13 on their latest conference. One of the upcoming features is the usage of [Server Components](https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components) in the `app` directory. So far, faustjs is using the provider pattern wrapping parts or even the whole app in `<FaustProvider>`. While this totally makes sense for the shared Apollo client etc., I'm wondering how we could use faust in the future leveraging both server components to reduce bundle size and some features of faust. Would this be even possible without using "use client" for all blog related pages? Or would we need to get rid of FaustProvider and use a hook, but how could the client be shared?

Thanks

**Comment by theodesp** [comment]
Hello @cobrani.
I hope you are doing well.
We have a ticket in our backlog to investigate this new feature. It would definitely means that we will have to propose new changes to the framework that align with our strategy while making use of the new improved features. We will post new updates in the blog.

**Comment by cobrani** [comment]
Hey @theodesp, thanks a lot for this insight! BTW regarding Faust.js, I could easily migrate to Next.js 13 without problems using the old pages structure (but didn't upgrade to new Faust.js using Apollo yet).
